src/Auth0.AuthenticationApi/Tokens/IdTokenRequirements.cs

```diff
@@ -41,8 +46,9 @@ internal class IdTokenRequirements
         /// <param name="leeway">Amount of leeway in validating date and time claims to allow some clock variance
         /// between the issuer and the application.</param>
         /// <param name="maxAge">Optional maximum time since the user last authenticated.</param>
-        public IdTokenRequirements(string issuer, string audience, TimeSpan leeway, TimeSpan ? maxAge = null)
+        public IdTokenRequirements(JwtSignatureAlgorithm signatureAlgorithm, string issuer, string audience, TimeSpan leeway, TimeSpan? maxAge = null)
```

Comment:
Should `leeway` be nullable here? I wouldn't require that anywhere if you can just fallback on the default.

Reply:
Originally the IdTokenRequirements class/code was shared between OIDC and Auth0.NET.  As one is meant for mobile/native and the other for servers there wasn't a "good" default that applied to both so it made sense to not specify one.

There's a chance at some point we'll extract this code out into shared library both projects can use (and perhaps other people) and again I don't think there's a one-size-fits-all default for that.